src/vs/workbench/contrib/inlineChat/browser/inlineChatWidget.ts

```diff
@@ -382,7 +393,9 @@ export class InlineChatWidget {
 		if (hasMessage) {
 			const sessionModel = this._chatMessageDisposables.add(new ChatModel(message.providerId, undefined, this._logService, this._chatAgentService, this._instantiationService));
 			const responseModel = this._chatMessageDisposables.add(new ChatResponseModel(message.message, sessionModel, undefined, undefined, message.requestId, !isIncomplete, false, undefined));
-			const viewModel = this._chatMessageDisposables.add(new ChatResponseViewModel(responseModel, this._logService));
+			this._responseViewModel = this._chatMessageDisposables.add(new ChatResponseViewModel(responseModel, this._logService));
```

Comment:
This pattern is a bit weird to me, the ChatResponseModel is not actually inside the ChatModel, so then calling `updateCodeBlockTextModels` with a viewModel which is not inside that chatViewModel feels wrong. We should try to make this easier to reuse in a more natural way outside of panel chat

Reply:
I am not sure who owns this, but agree it is not ideal and was tough to figure out. I am out for the rest of the iteration. cc @mjbvz 

I can look in April during debt week

Reply:
Maybe we can move the parsing logic `CodeBlockModelCollection` instead but keep the syncing stuff in `ChatViewModel`?

Reply:
just fyi that I working on a change to reuse the whole `ChatWidget` for inline chat. That makes this obsolete

Reply:
Perfect, I think that's exactly what @meganrogge was after